iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/CitationsActionBuilderModuleFactory.java

```diff
@@ -165,21 +165,20 @@ public CitationEntry validate(CitationEntry citationEntry, Float confidenceLevel
          * Allows to normalize a citation entry.
          */
         public static class CitationEntryNormalizer {
-            private CitationEntryMatchChecker citationEntryMatchChecker = new CitationEntryMatchChecker();
-            private MatchingResultCitationEntryNormalizer matchingResultCitationEntryNormalizer = new MatchingResultCitationEntryNormalizer();
+            private DestinationDocumentIdNormalizer destinationDocumentIdNormalizer = new DestinationDocumentIdNormalizer();
 
             public CitationEntry normalize(CitationEntry citationEntry) {
-                if (citationEntryMatchChecker.isMatchingResult(citationEntry)) {
-                    return matchingResultCitationEntryNormalizer.normalize(citationEntry);
+                if (Objects.nonNull(citationEntry.getDestinationDocumentId())) {
+                    return destinationDocumentIdNormalizer.normalize(citationEntry);
```

Comment:
Thanks but we could make it even more straightforward: since we are checking the `destinationDocumentId` for existence in order to apply a modification on it then I hardly see a point in applying the normalization on the whole `citationEntry` object which is quite confusing. 

I think there is no need to overcomplicate, we could easily skip the `DestinationDocumentIdNormalizer` layer accepting `citationEntry` and replace:

```
destinationDocumentIdNormalizer.normalize(citationEntry);
```
with the content of this unnecessary "wrapping" method:
```
citationEntry.setDestinationDocumentId(destinationDocumentIdNormalizerFn.apply(citationEntry.getDestinationDocumentId()));
```
You check the document id and you apply the modification on it straight away. Doesn't it look simpler?

Reply:
Done :)